boot/systems.go

```diff
@@ -447,3 +448,38 @@ func DropRecoverySystem(dev Device, systemLabel string) error {
 	const expectReseal = true
 	return resealKeyToModeenv(dirs.GlobalRootDir, m, expectReseal)
 }
+
+// AddRecoveryCapableSystem records a given system as one that we can recover
+// from.
+func AddRecoveryCapableSystem(systemLabel string) error {
+	opts := &bootloader.Options{
+		// setup the recovery bootloader
+		Role: bootloader.RoleRecovery,
+	}
+	// TODO:UC20: seed may need to be switched to RW
+	bl, err := bootloader.Find(InitramfsUbuntuSeedDir, opts)
+	if err != nil {
+		return err
+	}
+	rbl, ok := bl.(bootloader.RecoveryAwareBootloader)
+	if !ok {
+		return nil
+	}
+	vars, err := rbl.GetBootVars("snapd_recovery_capable_systems")
+	if err != nil {
+		return err
+	}
+	var systems []string
+	if vars["snapd_recovery_capable_systems"] != "" {
+		systems = strings.Split(vars["snapd_recovery_capable_systems"], ",")
+		if strutil.ListContains(systems, systemLabel) {
+			// already recorded, nothing to do
```

Comment:
we should probably shift it anyway at end or beginning depending on what we decide about my previous question

Reply:
this comment is still valid about moving it at the end I think

Reply:
with GoodRecoverySystems in modeenv, we append elements at the end, unless this is about something else and I misunderstood the comment, or do you mean that we should inspect the list and make sure that the systemLabel appears last in the list?

Reply:
this is done now